src/fromager/context.py

```diff
@@ -95,9 +95,19 @@ def __init__(
 
         self._parallel_builds = False
 
+        self._pre_built_override: set[NormalizedName] = set()
+
     def enable_parallel_builds(self) -> None:
         self._parallel_builds = True
 
+    def add_pre_built_override(self, package_name: str | NormalizedName) -> None:
```

Comment:
With the change in the logic in the bootstrapper to invoke the logic for prebuilt packages directly, I think you can remove this function.

Reply:
We use it through `_mark_package_as_pre_built_runtime(req)`  in _build_package()
```
 try:
                self._mark_package_as_pre_built_runtime(req)
                wheel_url, _ = self._resolve_prebuilt_with_history(
                    req=req, req_type=RequirementType.TOP_LEVEL
                )
                wheel_filename, unpack_dir = self._download_prebuilt(
                    req=req,
                    req_type=RequirementType.TOP_LEVEL,
                    resolved_version=resolved_version,
                    wheel_url=wheel_url,
                )
                logger.info(
                    "test mode: successfully handled %s as pre-built after build failure",
                    req.name,
                )
```

Reply:
Sorry, I misunderstood. I removed my previous comments. Going to fix it.